editoast/src/views/timetable/stdcm.rs

```diff
@@ -292,7 +294,10 @@ async fn stdcm(
             .collect(),
     };
 
-    let stdcm_response = stdcm_request.fetch(core_client.as_ref()).await;
+    let stdcm_response: Result<core::stdcm::Response, InternalError> = stdcm_request
+        .fetch(core_client.as_ref())
+        .await
+        .map_err(Into::into);
```

Comment:
Here, we convert the error side of `stdcm_response` into an `InternalError`... but only to transform it again L317 below into again something else (note, it's not part of your PR, the code was already like this before, it's just that looking at the modification, it highlighted it). Why not doing all the transformation right away, something like the following?

```suggestion
    let stdcm_response = stdcm_request
        .fetch(core_client.as_ref())
        .await
        .map(StdcmResponseOrError::Response)
        .map_err(|error| {
                let error: InternalError = error.into();
                StdcmResponseOrError::RequestError(serde_json::to_value(error).unwrap_or(
                    serde_json::Value::String("Failed to serialize the error".to_string()),
                ))
            });
```

You should be able to get rid of the lines below (317-325).

Reply:
Oh, after discussion, it is indeed not the same variable. The second one is scoped inside an `if` and the first one is still used after the `if`. So what I propose is not possible.